modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java

```diff
@@ -496,6 +513,85 @@ public void writeBitSet(BitSet val) {
         writeLongArray(val != null ? val.toLongArray() : null);
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public void writeByteBuffer(ByteBuffer val) {
+        switch (byteBufferState) {
+            case 0:
+                byte flag = 0;
+
+                if (val != null) {
+                    flag |= 1;
+
+                    if (val.order() == ByteOrder.BIG_ENDIAN) {
+                        flag |= 2;
+                    }
+
+                    if (val.position() == 0) {
+                        flag |= 4;
+                    }
+
+                    if (val.limit() == val.capacity()) {
```

Comment:
I'd prefer != here. So if limit is set, then set the flag

Reply:
Sure